Refactor handler.go

requilence opened the pull request:
- split `Run()` method into `Run()` and `RunOnce()`
- take different results processing methods(file/continious file/hub wait/hub interval) out of `RunOnce()` 

gocyclo:
```
19 frontman (*Frontman).processInput handler.go:342:1
17 frontman (*Frontman).HostInfoResults handler.go:470:1
15 frontman (*Frontman).PostResultsToHub handler.go:109:1
13 frontman (*Frontman).RunOnce handler.go:281:1
```

soupdiver commented:
I'm not done with review just had brief look.
I always criticised the amount of nesting of `if` and `else` branches we have. So maybe this can still be optimised: https://github.com/cloudradar-monitoring/frontman/blob/3788d207e3c2f41a8c918aee567626716303cd63/handler.go#L285

Just because `gocyclo` is under 20 now doesn't mean it's fine. It' just an indicator.

requilence commented:
Yes, I tried to optimize this as well, but because this is error handling it looks strange to make a separate method just to add the `auth` string only in one case. Don't you think?

requilence commented:
> The problem is again that the code ist just not obvious.
I made the auth logging more simple

soupdiver commented:
@requilence Are you speaking about https://github.com/cloudradar-monitoring/frontman/pull/18/commits/f83ed2163738af009beadf46df5a5e7d92730af3 ?
There are still no comments and the changes are so minor that things are not improved much in my opinion.

requilence commented:
just added some comments

requilence commented:
Picked out FetchInput() from RunOnce()

requilence commented:
Let me explain the idea behind this snippet

```
if err != nil {
			auth := ""
			if fm.HubUser != "" {
				// include Basic Auth login for the context if available
				auth = fmt.Sprintf(" (%s:***)", fm.HubUser)
			}
			return fmt.Errorf("InputFromHub%s: %s", auth, err.Error())
		}
```

When I implemented the Basic Auth, I added this code to add some auth context to the log. E.g. If I was rejected because of incorrect credentials

`time="2018-11-21T15:50:40+02:00" level=error msg="InputFromHub(user123:***): 401 Unauthorized"`.

I initially thought to remove this after the test but decided to left because it can be useful in other cases, e.g. `time="2018-11-21T15:51:13+02:00" level=error msg="InputFromHub(user123:***): 402 Payment Required"`

If you think this is unnecessary, lets remove this code

requilence commented:
I tried to make a compact comment with an example
Also, I swap the condition to decrease the nesting level

soupdiver commented on cmd/frontman/main.go:
```diff
@@ -338,12 +338,28 @@ sudo sysctl -w net.ipv4.ping_group_range="0   2147483647"`
+
+		err = fm.RunOnce(input, output, interruptChan, false)
+		if err != nil {
+			log.Error(err)
```
Why do we `return` in the error case before but not in this one?
If it's an indirect return done on line 359 please add a return into the error handling
